app/uk/gov/hmrc/bindingtariffclassification/service/CaseService.scala

```diff
@@ -40,13 +41,18 @@ class CaseService @Inject()(appConfig: AppConfig,
   def addInitialSampleStatusIfExists(c: Case): Future[Unit] = {
     if (c.sampleStatus.nonEmpty) {
       val details = SampleStatusChange(None, c.sampleStatus, None)
-      eventService.insert(Event(UUID.randomUUID().toString, details, Operator("-1",Some(c.application.contact.name)), c.reference, Instant.now()))
+      eventService.insert(Event(UUID.randomUUID().toString, details, Operator("-1", Some(c.application.contact.name)), c.reference, Instant.now()))
     }
     Future.successful((): Unit)
   }
 
-  def nextCaseReference: Future[String] = {
-    sequenceRepository.incrementAndGetByName("case").map(_.value + appConfig.caseReferenceStart).map(_.toString)
+  def nextCaseReference(applicationType: ApplicationType): Future[String] = {
+    sequenceRepository.incrementAndGetByName("case").map {
+      _.value + appConfig.caseReferenceStart + (applicationType match {
+        case ApplicationType.BTI => appConfig.btiReferenceOffset
```

Comment:
Because of the strange offset logic they want there will be a collision from the 300,000,000th Liability onwards. Matt doesn't think it will ever happen though (we'll see about that...)

Reply:
counter + (  WhereToStart + HowManyExtraOnTop )  ....  sounds a good fun